provider/sidecar.go

```diff
@@ -92,11 +106,8 @@ func (provider *Sidecar) Provide(configurationChan chan<- types.ConfigMessage, p
 	if err != nil {
 		log.Fatalln("Error reloading Sidecar config", err)
 	}
-	err = provider.loadSidecarConfig(states.ByService())
-	if err != nil {
-		return err
-	}
-	return nil
+
+	return provider.loadSidecarConfig(states.ByService())
```

Comment:
@relistan Changes pushed to address this... Hopefully, it will fit nicely with the Sidecar watcher additions :)